src/core/heavyGraph.js

```diff
@@ -0,0 +1,127 @@
+// @flow
```

Comment:
Looks like a great start, only nit is that the filename should also be updated to `weightedGraph2.js` instead of `heavyGraph.js`